README.md

```diff
@@ -1,20 +1,27 @@
-# Git-Service
-
-Git Service: Enhanced Git Navigation Tool
-
-## Prerequisites
-
+# Git-Service: Enhanced Git Navigation Tool
+
+## Purpose
+It is hard for developers to track branches and tags in GitHub. Especially in an active work group where developers collaborate and work on multiple branches, cherry-pick, merge, rebase and patch operations make it even more difficult to search for a commit. Searching for a commit may be required in cases like code review, debugging, reverting specific changes, etc. Moreover, lots of continuous integration jobs are triggered by events or run periodically every day. Hence, finding out relationships between commits and jobs is also a tough task. This project is crucial because it addresses these challenges by introducing a tool that simplifies navigating through commits, branches, tags, and clarifies the connections between the code changes and CI/CD processes.
+
+## Features
+1. `/<owner>/<repo>/branch/getActiveBranches`: Provides a list of branches that have been active in the given time frame
+2. `/<owner>/<repo>/branch/getBranchByTag`: Provides the name of the branch from which the given tag was built
+3. `/<owner>/<repo>/tag/getChildTagsByCommit`: Provides the child tags in each branch
+4. `/<owner>/<repo>/tag/getParentTagsByCommit`: Provides the nearest parent tags in each branch
+5. `/<owner>/<repo>/commit/getCommitsBefore`: Provides the commits before the given commit in the branches it is present in
+6. `/<owner>/<repo>/commit/getCommitsAfter`: Provides the commits after the given commit in the branches it is present in
+7. `/<owner>/<repo>/commit/getCommitByName`: Provides list of commits when searched by name
+8. `/<owner>/<repo>/commit/getCommitByDescription`: Provides list of commits when searched by description
+9. `/<owner>/<repo>/commit/getCommitByAuthor`: Provides list of commits when searched by author
+10. `/<owner>/<repo>job/getJobsByCommit`: Provides the CI/CD jobs associated with the given commit
+11. `/<owner>/<repo>/commit/commitReleased`: Accepts commit id and release branch name. Tells if the commit is released or not
+
+
+## Prerequisites: Installation Instructions
 * Install the `golang` compiler from the [official source](https://go.dev) (version 1.21.3)
 * Include $GOPATH in your path via `export PATH=$PATH:$(go env GOPATH)/bin`
 * Adding the above to your `~/.bashrc` file will require `source ~/.bashrc`
 
```

Comment:
* Alternatively, if using a package manager like homebrew for example, install golang by running `brew install golang`

Reply:
Thank you for the review, I added it to readme file